### PR TITLE
tailfs: fix startup issues on windows

### PR DESCRIPTION
**Pull request by oxtoacart**
Starts TailFS for Windows too, initializes shares on startup.

Updates tailscale/corp#16827

### CODE REVIEW

**Comment by oxtoacart** [comment]
This is required to allow the Windows app to start successfully.

File: cmd/tailscaled/tailscaled.go [hunk 1]
```diff
@@ -403,14 +403,16 @@ func run() (err error) {
...
-
 	return startIPNServer(context.Background(), logf, pol.PublicID, sys)
 }
 
 var sigPipe os.Signal // set by sigpipe.go
 
 func startIPNServer(ctx context.Context, logf logger.Logf, logID logid.PublicID, sys *tsd.System) error {
+	// Set up tailfs.FileSystemForRemote here so that we get it on both Windows
+	// and UNIX.
+	sys.Set(tailfsimpl.NewFileSystemForRemote(logf))
```
Comment (bradfitz) [inline cmd/tailscaled/tailscaled.go:414]: We also get it on iOS. Are we going to be a file server there too?

**Review by maisem — APPROVED** [review]

**Review by bradfitz — COMMENTED** [review]
s/startup fixes/fix startup on Windows/

or "fix issues on startup"

("startup fixes" is an adjective and a noun without a verb and we conventionally start with a verb)

**Comment by maisem** [comment]
I don't fully understand this, I assume you've tested it

File: cmd/tailscaled/tailscaled.go (hunk 1 above)
Comment (oxtoacart) [reply cmd/tailscaled/tailscaled.go:414]: No, we don't want to be a file server there. Let me fix.

Comment (oxtoacart) [reply cmd/tailscaled/tailscaled.go:414]: Actually, I don't see us importing `cmd/tailscaled` on iOS. What makes you say we'll get this on IOS?

File: cmd/tailscaled/tailscaled_windows.go [hunk 2]
```diff
@@ -315,6 +316,8 @@ func beWindowsSubprocess() bool {
 	}
 	sys.Set(netMon)
 
+	sys.Set(tailfsimpl.NewFileSystemForRemote(log.Printf))
```
Comment (oxtoacart) [inline cmd/tailscaled/tailscaled_windows.go:319]: @bradfitz to be safe, I'm just doing this explicitly on the Windows codepath.

**Pull request merged**